temporalio/failure.py

```diff
@@ -0,0 +1,61 @@
+from enum import IntEnum
+from typing import Any, List, Optional
+
+import temporalio.api.enums.v1
+import temporalio.api.failure.v1
+import temporalio.converter
+
+
+class FailureError(Exception):
+    @staticmethod
+    async def from_proto(
+        failure: temporalio.api.failure.v1.Failure,
+        data_converter: temporalio.converter.DataConverter,
+    ) -> "FailureError":
+        raise NotImplementedError
+
+    def __init__(self, message: str, failure: "Failure") -> None:
+        super().__init__(message)
+        raise NotImplementedError
+
+
+class Failure:
+    def __init__(
+        self,
+        *details: Any,
+        proto_failure: Optional[temporalio.api.failure.v1.Failure] = None
+    ) -> None:
+        raise NotImplementedError
+
+
+class CancelledFailure(Failure):
+    def __init__(self, *details: Any) -> None:
+        super().__init__(*details)
+        raise NotImplementedError
+
+
+class TerminatedFailure(Failure):
+    def __init__(self, *details: Any, reason: Optional[str]) -> None:
```

Comment:
Are there `details` in all of these failures?

Reply:
There will be, notice the `NotImplementedError`. There will also be docs. And tests.

Reply:
I have added an `exceptions` module (not yet documented or completely tested)